contributed_definitions/nyaml/NXphotoemission.yaml

```diff
@@ -0,0 +1,1641 @@
+category: application
+doc: |
+  This is the most general application definition for
+  photoemission experiments.
+  
+  Groups and fields are named according to the
+  `ISO 18115-1:2023`_ specification as well as the `IUPAC Recommendations 2020`_.
+  
+  .. _ISO 18115-1:2023: https://www.iso.org/standard/74811.html
+  .. _IUPAC Recommendations 2020: https://doi.org/10.1515/pac-2019-0404
+symbols:
+  doc: |
+    The symbols used in the schema to specify e.g. dimensions of arrays
+  n_transmission_function: |
+    Number of data points in the transmission function.
+type: group
+NXphotoemission(NXobject):
+  (NXentry):
+    definition:
+      \@version:
+      enumeration: [NXphotoemission]
```

Comment:
This enumeration should probably contain all derived application definitions as well, no? So that an NXmpes file ideed can be validated as an NXphotoemission file.

Reply:
I'm not sure if this is a good idea. But I think it's a very good remark as I think this is still a problem with the inheritance approach.
I understand your point that a consuming application would want to know if it is able to read a definition. Currently, it effectively needs to know the full inheritance tree of NXphotoemission and therefore a copy of the definitions or a tool like pynxtools to ask for the info.

From a language standpoint, however, I would say that adding multiple names here just act as a synonym for this definition. I need to context of the sub-appdef to know how this should be interpreted but I would argue that the definitions (at least their language) should be context-free if I don't specify the context explicitly (like in inheritance).

I think a better way to go would be to request to include the full inheritance chain into the file, e.g., for NXmpes_arpes we need somewhere the chain `NXmpes_arpes` -> `NXmpes` -> `NXphotoemission` -> `NXobject` stored. That way a tool can just search for a matching definition and knows that it should be fulfilled by the inheritance chain.